### PR TITLE
[FLINK-8475][config][docs] Integrate TM options

### PR DESCRIPTION
**Pull request by zentol**
## What is the purpose of the change

This PR integrates the TaskManager `ConfigOptions` into the configuration docs generator.

## Brief change log

* Port missing  taskmanager config constants to config options
* Add missing descriptions to config options (derived from existing description/javadocs)
* integrate TaskManager configuration table into `config.md`

### CODE REVIEW

File: docs/ops/config.md [hunk 1]
```diff
@@ -268,59 +268,7 @@ These parameters configure the default HDFS used by Flink. Setups that do not sp
...
-
-- `taskmanager.tmp.dirs`: The directory for temporary files, or a list of directories separated by the system's directory delimiter (for example ':' (colon) on Linux/Unix). If multiple directories are specified, then the temporary files will be distributed across the directories in a round robin fashion. The I/O manager component will spawn one reading and one writing thread per directory. A directory may be listed multiple times to have the I/O manager use multiple threads for it (for example if it is physically stored on a very fast disc or RAID) (DEFAULT: **The system's tmp dir**).
-
-- `taskmanager.network.memory.fraction`: Fraction of JVM memory to use for network buffers. This determines how many streaming data exchange channels a TaskManager can have at the same time and how well buffered the channels are. If a job is rejected or you get a warning that the system has not enough buffers available, increase this value or the min/max values below. Also note, that `taskmanager.network.memory.min` and `taskmanager.network.memory.max` may override this fraction. (DEFAULT: **0.1**)
```
Comment (aljoscha) [inline docs/ops/config.md]: Why not include the longer description?

Comment (aljoscha) [reply docs/ops/config.md]: This question is probably true for a lot of these changes.

File: docs/ops/config.md [hunk 2]
```diff
@@ -268,59 +268,7 @@ These parameters configure the default HDFS used by Flink. Setups that do not sp
...
-
-- `taskmanager.network.memory.min`: Minimum memory size for network buffers in bytes (DEFAULT: **64 MB**). Previously, this was determined from `taskmanager.network.numberOfBuffers` and `taskmanager.memory.segment-size`.
-
-- `taskmanager.network.memory.max`: Maximum memory size for network buffers in bytes (DEFAULT: **1 GB**). Previously, this was determined from `taskmanager.network.numberOfBuffers` and `taskmanager.memory.segment-size`.
-
-- `taskmanager.network.numberOfBuffers` (deprecated, replaced by the three parameters above): The number of buffers available to the network stack. This number determines how many streaming data exchange channels a TaskManager can have at the same time and how well buffered the channels are. If a job is rejected or you get a warning that the system has not enough buffers available, increase this value (DEFAULT: **2048**). If set, it will be mapped to `taskmanager.network.memory.min` and `taskmanager.network.memory.max` based on `taskmanager.memory.segment-size`.
```
Comment (aljoscha) [inline docs/ops/config.md]: Isn't this deprecated but you can still use it? I don't see it in the new doc.

File: docs/ops/config.md [hunk 3]
```diff
@@ -268,59 +268,7 @@ These parameters configure the default HDFS used by Flink. Setups that do not sp
...
-
-- `taskmanager.memory.fraction`: The relative amount of memory (with respect to `taskmanager.heap.mb`, after subtracting the amount of memory used by network buffers) that the task manager reserves for sorting, hash tables, and caching of intermediate results. For example, a value of `0.8` means that a task manager reserves 80% of its memory (on-heap or off-heap depending on `taskmanager.memory.off-heap`) for internal data buffers, leaving 20% of free memory for the task manager's heap for objects created by user-defined functions. (DEFAULT: 0.7) This parameter is only evaluated, if `taskmanager.memory.size` is not set.
-
-- `taskmanager.debug.memory.startLogThread`: Causes the TaskManagers to periodically log memory and Garbage collection statistics. The statistics include current heap-, off-heap, and other memory pool utilization, as well as the time spent on garbage collection, by heap memory pool.
```
Comment (aljoscha) [inline docs/ops/config.md]: It looks like this and the next config didn't make it into the new doc

File: docs/ops/config.md [hunk 4]
```diff
@@ -268,59 +268,7 @@ These parameters configure the default HDFS used by Flink. Setups that do not sp
...
-
-- `taskmanager.debug.memory.logIntervalMs`: The interval (in milliseconds) in which the TaskManagers log the memory and garbage collection statistics. Only has an effect, if `taskmanager.debug.memory.startLogThread` is set to true.
-
-- `taskmanager.maxRegistrationDuration`: Defines the maximum time it can take for the TaskManager registration. If the duration is exceeded without a successful registration, then the TaskManager terminates. The max registration duration requires a time unit specifier (ms/s/min/h/d) (e.g. "10 min"). (DEFAULT: **Inf**)
-
-- `taskmanager.initial-registration-pause`: The initial registration pause between two consecutive registration attempts. The pause is doubled for each new registration attempt until it reaches the maximum registration pause. The initial registration pause requires a time unit specifier (ms/s/min/h/d) (e.g. "5 s"). (DEFAULT: **500 ms**)
-
-- `taskmanager.max-registration-pause`: The maximum registration pause between two consecutive registration attempts. The max registration pause requires a time unit specifier (ms/s/min/h/d) (e.g. "5 s"). (DEFAULT: **30 s**)
```
Comment (aljoscha) [inline docs/ops/config.md]: This seems to not be in the new doc

File: docs/ops/config.md [hunk 5]
```diff
@@ -268,59 +268,7 @@ These parameters configure the default HDFS used by Flink. Setups that do not sp
...
-
-- `taskmanager.refused-registration-pause`: The pause after a registration has been refused by the job manager before retrying to connect. The refused registration pause requires a time unit specifier (ms/s/min/h/d) (e.g. "5 s"). (DEFAULT: **10 s**)
-
-- `taskmanager.jvm-exit-on-oom`: Indicates that the TaskManager should immediately terminate the JVM if the task thread throws an `OutOfMemoryError` (DEFAULT: **false**).
-
-- `blob.fetch.retries`: The number of retries for the TaskManager to download BLOBs (such as JAR files) from the JobManager (DEFAULT: **50**).
-
-- `blob.fetch.num-concurrent`: The number concurrent BLOB fetches (such as JAR file downloads) that the JobManager serves (DEFAULT: **50**).
```
Comment (aljoscha) [inline docs/ops/config.md]: Where were those migrated to?

File: docs/_includes/generated/task_manager_configuration.html [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    <thead>
+        <tr>
+            <th class="text-left" style="width: 20%">Key</th>
+            <th class="text-left" style="width: 15%">Default</th>
+            <th class="text-left" style="width: 65%">Description</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td><h5>task.cancellation.interval</h5></td>
```
Comment (aljoscha) [inline docs/_includes/generated/task_manager_configuration.html]: The description for those two should probably be carried over from the old doc

File: docs/ops/config.md [hunk 7]
```diff
@@ -268,59 +268,7 @@ These parameters configure the default HDFS used by Flink. Setups that do not sp
...
-
-- `blob.fetch.backlog`: The maximum number of queued BLOB fetches (such as JAR file downloads) that the JobManager allows (DEFAULT: **1000**).
-
-- `task.cancellation-interval`: Time interval between two successive task cancellation attempts in milliseconds (DEFAULT: **30000**).
```
Comment (aljoscha) [inline docs/ops/config.md]: You know why the new option is `task.cancellation.interval`?

File: docs/ops/config.md [hunk 8]
```diff
@@ -268,59 +268,7 @@ These parameters configure the default HDFS used by Flink. Setups that do not sp
...
-
-- `taskmanager.numberOfTaskSlots`: The number of parallel operator or user function instances that a single TaskManager can run (DEFAULT: **1**). If this value is larger than 1, a single TaskManager takes multiple instances of a function or operator. That way, the TaskManager can utilize multiple CPU cores, but at the same time, the available memory is divided between the different operator or function instances. This value is typically proportional to the number of physical CPU cores that the TaskManager's machine has (e.g., equal to the number of cores, or half the number of cores).
-
-- `taskmanager.tmp.dirs`: The directory for temporary files, or a list of directories separated by the system's directory delimiter (for example ':' (colon) on Linux/Unix). If multiple directories are specified, then the temporary files will be distributed across the directories in a round robin fashion. The I/O manager component will spawn one reading and one writing thread per directory. A directory may be listed multiple times to have the I/O manager use multiple threads for it (for example if it is physically stored on a very fast disc or RAID) (DEFAULT: **The system's tmp dir**).
```
[the remark this answers is not in the document]
Comment (zentol) [reply docs/ops/config.md]: superceeded by `CoreOptions#TMP_DIRS`

File: docs/ops/config.md (hunk 2 above)
Comment (zentol) [reply docs/ops/config.md]: the generator excludes deprecated options on the assumption that they were subsumed by other options.

File: docs/ops/config.md (hunk 3 above)
Comment (zentol) [reply docs/ops/config.md]: weren't ported to ConfigOptions yet.

File: docs/ops/config.md (hunk 4 above)
Comment (zentol) [reply docs/ops/config.md]: yes missed this one

File: docs/ops/config.md (hunk 5 above)
Comment (zentol) [reply docs/ops/config.md]: BlobServerOptions, see #5391

File: docs/ops/config.md (hunk 7 above)
Comment (zentol) [reply docs/ops/config.md]: old key was deprecated
```
public static final ConfigOption<Long> TASK_CANCELLATION_INTERVAL =
			key("task.cancellation.interval")
			.defaultValue(30000L)
			.withDeprecatedKeys("task.cancellation-interval");
```

**Comment by zentol** [comment]
man these long descriptions are gonna wreck the table...

File: docs/_includes/generated/task_manager_configuration.html (hunk 6 above)
Comment (zentol) [reply docs/_includes/generated/task_manager_configuration.html]: there's a description for cancellation.interval, which other one are you referring to?

File: docs/ops/config.md [hunk 9]
```diff
@@ -268,59 +268,7 @@ These parameters configure the default HDFS used by Flink. Setups that do not sp
...
-
-- `taskmanager.rpc.port`: The task manager's IPC port (DEFAULT: **0**, which lets the OS choose a free port). Flink also accepts a list of ports ("50100,50101"), ranges ("50100-50200") or a combination of both. It is recommended to set a range of ports to avoid collisions when multiple TaskManagers are running on the same machine.
-
-- `taskmanager.data.port`: The task manager's port used for data exchange operations (DEFAULT: **0**, which lets the OS choose a free port).
-
-- `taskmanager.data.ssl.enabled`: Enable SSL support for the taskmanager data transport. This is applicable only when the global ssl flag security.ssl.enabled is set to true (DEFAULT: **true**)
-
-- `taskmanager.heap.mb`: JVM heap size (in megabytes) for the TaskManagers, which are the parallel workers of the system. In contrast to Hadoop, Flink runs operators (e.g., join, aggregate) and user-defined functions (e.g., Map, Reduce, CoGroup) inside the TaskManager (including sorting/hashing/caching), so this value should be as large as possible (DEFAULT: **512**). On YARN setups, this value is automatically configured to the size of the TaskManager's YARN container, minus a certain tolerance value.
```
[the remark this answers is not in the document]
Comment (zentol) [reply docs/ops/config.md]: Actually yes, I believe this doesn't belong in there. The YARN note is useful though...

**Comment by zentol** [comment]
@aljoscha I've ported the missing options and modified plenty of descriptions.

File: docs/_includes/generated/task_manager_configuration.html (hunk 6 above)
Comment (aljoscha) [reply docs/_includes/generated/task_manager_configuration.html]: `cancellation.timeout`, I think I was referring to

**Comment by aljoscha** [comment]
Feel free to merge once you addressed my other comment. I'm eager to merge this no matter what because all of this is better than what we had before...

**Comment by zentol** [comment]
merging.

**Pull request closed without merging**